src/libraries/Directory.Build.targets

```
@@ -240,6 +237,22 @@
     <ExcludeFromPackage Condition="'$(TargetFramework)' == '$(NetFrameworkCurrent)' and '$(ExcludeCurrentFullFrameworkFromPackage)' == 'true'">true</ExcludeFromPackage>
   </PropertyGroup>
 
+  <!-- Adds System.Runtime.Versioning*Platform* annotation attributes to < .NET 5 builds -->
+  <Choose>
+    <When Condition="('$(IncludePlatformAttributes)' == 'true' or '$(IsWindowsSpecific)' == 'true') and ('$(TargetFrameworkIdentifier)' != '.NETCoreApp' or $(TargetFramework.StartsWith('netcoreapp')))">
```

No need to read from TargetFramework:

```suggestion
    <When Condition="('$(IncludePlatformAttributes)' == 'true' or '$(IsWindowsSpecific)' == 'true') and '$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
```

---

Wouldn't this break for `.net5.0`? The TFI of `.net5.0` is `.NETCoreApp` -- basically what this is trying to do is to include it only on .net5.0 or latter.

---

Ah I see. Then this should be:

```suggestion
    <When Condition="('$(IncludePlatformAttributes)' == 'true' or '$(IsWindowsSpecific)' == 'true') and ('$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0')))">
```

---

Great, didn't know there was `TargetFrameworkVersion`. 

This won't add the attributes to netstandard or netfx builds. I think we should just do: 

```suggestion
    <When Condition="('$(IncludePlatformAttributes)' == 'true' or '$(IsWindowsSpecific)' == 'true') and  $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0'))">
```

---

The comment says `  <!-- Adds System.Runtime.Versioning*Platform* annotation attributes to < .NET 5 builds -->
` so I assumed this should only be added to .NETCoreApp < 5.

---

This is more correct :)

```suggestion
    <When Condition="('$(IncludePlatformAttributes)' == 'true' or '$(IsWindowsSpecific)' == 'true') and ('$(TargetFrameworkIdentifier)' != '.NETCoreApp' or $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0')))">
```

---

Yeah, I didn't include `TargetFrameworkIdentifier != .NETCoreApp` cause we don't build any TFM that can have TFV higher or equal than 5.0, right?

---

right

---

> <!-- Adds System.Runtime.Versioning*Platform* annotation attributes to < .NET 5 builds --

I wanted to include this file in an explicit way for all TFMs that existed prior to .NET 5. I wanted to keep the condition as short as possible, I might go too far ;)

> $([MSBuild]::VersionLessThan

The last time I wanted to use it, I had to update VS to the latest preview. If possible, I would prefer to avoid using it.

@ViktorHofer @safern do you think that this is a common pattern? Perhaps we should introduce a new property called "IsNet5OrNewer" in some common `.props` file, add a comment there, and reuse it in places that need it?

---

adding the comment again as previous one had unclosed xml comment and look empty (click edit to see the content)

> <!-- Adds System.Runtime.Versioning*Platform* annotation attributes to < .NET 5 builds -->

I wanted to include this file in an explicit way for all TFMs that existed prior to .NET 5. I wanted to keep the condition as short as possible, I might go too far ;)

> $([MSBuild]::VersionLessThan

The last time I wanted to use it, I had to update VS to the latest preview. If possible, I would prefer to avoid using it.

@ViktorHofer @safern do you think that this is a common pattern? Perhaps we should introduce a new property called "IsNet5OrNewer" in some common `.props` file, add a comment there, and reuse it in places that need it?

---

Unfortunately we can't add such a property to the props files as the TargetFrameworkVersion and TargetFrameworkIdentifier aren't available until the project is evaluate, means not available before targets.

---

> Unfortunately we can't add such a property to the props files as the TargetFrameworkVersion and TargetFrameworkIdentifier aren't available until the project is evaluate, means not available before targets.

thanks for writing that! You have just saved few hours of my life spent for cursing MSBuild ;)

---

> I wanted to keep the condition as short as possible, I might go too far ;)

Please use the long form that I proposed above just to avoid issues with a (not planned) .NET Standard 5 version.